File: src/hetzner_code/naming.py
from __future__ import annotations

import secrets


def generate_instance_name(repo_name: str) -> str:
    return f"{repo_name}-{secrets.token_hex(3)}"


def generate_key_title(instance_name: str, repo_name: str) -> str:
    return f"hcode-{instance_name}-{repo_name}-{secrets.token_hex(3)}"

File: internal/naming/naming.go
// Package naming generates instance and deploy-key names.
package naming

import (
	"crypto/rand"
	"encoding/hex"
	"fmt"
)

func tokenHex(nbytes int) string {
	b := make([]byte, nbytes)
	if _, err := rand.Read(b); err != nil {
		panic(err)
	}
	return hex.EncodeToString(b)
}

// GenerateInstanceName builds "<repo>-<random>".
func GenerateInstanceName(repoName string) string {
	return fmt.Sprintf("%s-%s", repoName, tokenHex(3))
}

// GenerateKeyTitle builds "hcode-<instance>-<repo>-<random>".
func GenerateKeyTitle(instanceName, repoName string) string {
	return fmt.Sprintf("hcode-%s-%s-%s", instanceName, repoName, tokenHex(3))
}
